modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFRequestForm/steps/MdfList/MDFList.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import Table from './components/Table';
+import MDFListColumn from './components/MDFListColumn';
+import Button from '@clayui/button';
+import MDFRequest from '../../../../common/interfaces/mdfRequest';
+import MDFRequestStepProps from '../../interfaces/mdfRequestStepProps';
+
+const MdfList: any = ({
+	onNewRequest
+}:MDFRequestStepProps<MDFRequest>) => {
+
+	const {data} = useGetMDFRequest();
+
+	const mdfListObject = data?.items.reduce(
+		(objAccumulator: any, values: any) => {
+
+			const obj = {
+				activityPeriod: values.minDateActivity + values.maxDateActivity,
+				approved: '',
+				reimpursementClaim: '',
+				requestId: values.id,
+				requested: '',
+				status: values.status.label_i18n,
+				totalCost: '',
+			
+		};
+
+			return {
+ 
+				items: [...objAccumulator.items, obj],
+
+			};
+		},
+		{items: []}
+	);
```

Comment:
It should be good add a custom hook to wrapper this logic and use the reduce with useMemo.
- Let's try avoid to use any.
- Let's try avoid to use short names for variables.
- Maybe a map it's better then a reduce.